src/neptune/new/handler.py

```diff
@@ -253,43 +257,120 @@ def log(
+                attr = self._container.get_attribute(self._path)
+                if not attr:
+                    if is_collection(value):
+                        if value:
+                            first_value = next(iter(value))
+                        else:
+                            raise ValueError("Cannot deduce value type: `value` cannot be empty")
+                    else:
+                        first_value = value
+                    attr = self._create_new_attribute(first_value)
+                    self._container.set_attribute(self._path, attr)
+                attr.log(value, step=step, timestamp=timestamp, wait=wait, **kwargs)
+
+    def append(
+        self,
+        value: Union[dict, Any],
+        step: Optional[float] = None,
+        timestamp: Optional[float] = None,
+        wait: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        todo: NPT-12530
         """
         verify_type("step", step, (int, float, type(None)))
         verify_type("timestamp", timestamp, (int, float, type(None)))
-
         with self._container.lock():
             attr = self._container.get_attribute(self._path)
             if not attr:
                 if is_collection(value):
-                    if value:
-                        first_value = next(iter(value))
-                    else:
-                        raise ValueError("Cannot deduce value type: `value` cannot be empty")
-                else:
-                    first_value = value
-
-                if is_float(first_value):
-                    attr = FloatSeries(self._container, parse_path(self._path))
-                elif is_string(first_value):
-                    attr = StringSeries(self._container, parse_path(self._path))
-                elif FileVal.is_convertable(first_value):
-                    attr = FileSeries(self._container, parse_path(self._path))
-                elif is_float_like(first_value):
-                    attr = FloatSeries(self._container, parse_path(self._path))
-                elif is_string_like(first_value):
-                    warn_once(
-                        message="The object you're logging will be implicitly cast to a string."
-                        " We'll end support of this behavior in `neptune-client==1.0.0`."
-                        " To log the object as a string, use `.log(str(object))` instead.",
-                        stack_level=3,
-                    )
-                    attr = StringSeries(self._container, parse_path(self._path))
-                else:
-                    raise TypeError("Value of unsupported type {}".format(type(first_value)))
-
+                    raise ValueError("Value cannot be a collection")
+                attr = self._create_new_attribute(value)
                 self._container.set_attribute(self._path, attr)
             attr.log(value, step=step, timestamp=timestamp, wait=wait, **kwargs)
 
+    def _create_new_attribute(self, value):
```

Comment:
This method should be moved to `MetadataContainer` to be consistent with existing `define` used in `assign`.

Reply:
similar to assign - check it